Elaborate on the fixed mapping between proxy and request resources (#95)
ietf-wg-ohai/oblivious-http: draft-ietf-ohai-ohttp.md

# Operational and Deployment Considerations {#deployment}

This section discusses various operational and deployment considerations.

## Performance Overhead

Using Oblivious HTTP adds both cryptographic and latency to requests relative to
a simple HTTP request-response exchange.  Deploying proxy services that are on
path between clients and servers avoids adding significant additional delay due
to network topology.  A study of a similar system {{ODoH}} found that deploying
proxies close to servers was most effective in minimizing additional latency.

> ## Oblivious Proxy State {#proxy-state}

Reviewer: ```suggestion
## Resource Mappings {#proxy-state}
```

You might also change the anchor name to "mappings" or something to match.

> The Oblivious Proxy Resource is a fixed mapping between URI and the Oblivious
> Request Resource. This means that any encapsulated request sent to the Oblivious

Reviewer: ```suggestion
This protocol assumes a fixed, one-to-one mapping between the Oblivious Proxy Resource and 
the Oblivious Request Resource. This means that any encapsulated request sent to the Oblivious
```

Proxy Resource will always be forwarded to the Oblivious Request Resource.
This constraint was imposed to simplify proxy configuration and mitigate against
the Oblivious Proxy Resource being used as a generic proxy for unknown Oblivious
Request Resources. The proxy will only forward for Oblivious Request Resources
that it has explicitly configured and allowed.

> It is possible for an Oblivious Proxy to be configured with multiple Oblivious

Reviewer: "Oblivious Proxy" isn't really a concept we use.

```suggestion
It is possible for a server to be configured with multiple Oblivious
```

Author: We use it on and off. I'll remove other references to it in the doc.

Proxy Resources, each for a different Oblivious Request Resource as needed.

> As a consequence, the amount of state required for these mappings is linear
> in the number of Oblivious Proxy Resources supported.

Reviewer: This isn't strictly true as https://proxy.example/forward/example.com/foo -> https://example.com/foo and https://proxy.example/forward/example.com/bar -> https://example.com/bar demonstrates that you can embed all the necessary state in the URI.  Those are different proxy resources, but there is no additional state.

```suggestion
```

Author: Fair, yeah. Do we want to use this as an example configuration?

Reviewer: Probably better to use https://proxy.example/opr/a -> https://example.com/orr/foo instead or something like that.

No point telling people how to circumvent the 1:1 mapping so directly.

Author: Let's just omit the example for now then. We can add one later if folks think it's useful.
